File: staylio_frontend/common/interfaces/response/ChatSessionResponse.ts
import { ChatSessionStatus } from "@common/enums/ChatSessionStatus";
import { ChatSessionType } from "@common/enums/ChatSessionType";

export interface ChatSessionResponse {
    id: number;
    type: ChatSessionType;
    status: ChatSessionStatus;
    lastMessage: string;
    lastMessageAt: string;
    unreadCount: number;
    createdAt: string;

    // Customer Information
    customerId?: number;
    customerName?: string;
    customerAvatar?: string;

    // Branch Information
    branchId?: number;
}

File: staylio_backend/src/main/java/com/example/staylio_backend/dto/response/ChatSessionResponse.java
package com.example.staylio_backend.dto.response;

import com.example.staylio_backend.model.enums.ChatSessionStatus;
import com.example.staylio_backend.model.enums.ChatSessionType;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatSessionResponse {
    private Long id;

    private ChatSessionType type;

    private ChatSessionStatus status;

    private String lastMessage;

    private LocalDateTime lastMessageAt;

    private Long unreadCount;

    private LocalDateTime createdAt;
}
